pom.xml

```diff
@@ -73,6 +73,11 @@
         <artifactId>opentest4j</artifactId>
         <version>${opentest4j.version}</version>
       </dependency>
+      <dependency>
+        <groupId>org.junit.jupiter</groupId>
+        <artifactId>junit-jupiter-api</artifactId>
+        <version>${junit.version}</version>
```

Comment:
@jfgreffier @yury-s will this force the user to use the junit version that playwright expects?  Maybe we needs some documentation about excluding it if the user runs into conflicts with the version they are using?

We make this `provided` scope but then it may not build for a user that is not using junit and using testng?

Reply:
> Maybe we needs some documentation about excluding it if the user runs into conflicts with the version they are using?

Good point! We don't Playwright to dictate which exact JUnit version must be used in the project, I agree taht `<scope>provided</scope` would be appropriate here. We'll likely have a similar package for TestNG eventually and I assume the clients will either use one or the other and will have only one of the frameworks installed (or none, in case of scraping). We should work in those scenarios.

> We could make this provided scope but then it may not build for a user that is not using junit and using testng?

Correct me if I'm wrong, but unless the user explicitly imports one of the classes that depend on junit stuff, their project should work just fine without junit being available. Ot the same time, if their code does use junit, they are supposed to have junit library added to their deps. But I agree that it will be more tricky to document than what we have today.

As an alternative, we can put JUnit (and TestNG for that matter) related classes into their own Maven module, so that clients that use junit could import something like playwright-junit and those with TestNG playwright-testng. 

Interestingly enough we've managed to avoid all this dependency issue so far with playwright assertions.

Reply:
you're correct, just did a quick test and provided scope will solve this.